aea/cli/generate_all_protocols.py

```diff
@@ -0,0 +1,542 @@
+    current_specification_content = get_protocol_specification_from_readme(package_path)
+
+    # download latest protocol id and extract its protocol specification as above
+    configuration: ProtocolConfig = cast(
+        ProtocolConfig,
+        load_component_configuration(ComponentType.PROTOCOL, package_path),
+    )
+    temp_directory = Path(tempfile.mkdtemp())
+    try:
+        download_package(configuration.package_id, str(temp_directory))
+    except click.ClickException:
+        log("Protocol specification id not bumped - new protocol.")
+        return
+    downloaded_package_directory = temp_directory / configuration.name
+    old_specification_content = get_protocol_specification_from_readme(
+        downloaded_package_directory
+    )
+    old_configuration: ProtocolConfig = cast(
+        ProtocolConfig,
+        load_component_configuration(
+            ComponentType.PROTOCOL,
+            downloaded_package_directory,
+            skip_consistency_check=True,
+        ),
+    )
+
+    # if different, bump protocol specification version, else don't.
+    public_id_version_is_newer = (
+        old_configuration.public_id.package_version  # type: ignore
+        <= configuration.public_id.package_version
+    )
+    content_is_different = current_specification_content != old_specification_content
+    if public_id_version_is_newer and content_is_different:
+        log(
+            f"Bumping protocol specification id from '{old_configuration.protocol_specification_id}' to '{configuration.protocol_specification_id}'"
+        )
+        _bump_protocol_specification_id(package_path, configuration)
+        return
+    log(
+        "Protocol specification id not bumped - content is not different, or version is not newer."
+    )
+
+
+def _generate_all_protocols(packages_dir: Path, no_bump: bool = False) -> None:
+    """
+    Run the script.
+
+    :param packages_dir: Path to package dir
+    :param no_bump: if True, the (default: False)
+    """
+    _check_preliminaries()
+
+    all_protocols = list(find_protocols_in_local_registry(packages_dir))
+
+    with AEAProject():
+        log("=" * 100)
+        _process_test_protocol(
+            TEST_DATA / "sample_specification.yaml",
+            TEST_DATA / "generator" / "t_protocol",
+        )
+        log("=" * 100)
+        _process_test_protocol(
+            TEST_DATA / "sample_specification_no_custom_types.yaml",
+            TEST_DATA / "generator" / "t_protocol_no_ct",
+        )
+        for package_path in all_protocols:
+            log("=" * 100)
+            log(f"Processing protocol at path {package_path}")
+            if not no_bump:
+                _bump_protocol_specification_id_if_needed(package_path)
+            # no_bump implies to ignore the docstring:
+            #  'It was created with protocol buffer compiler ... and aea version ...'
+            _process_packages_protocol(package_path, no_bump)
```

Comment:
So how do we wanna approach this situation? Should generating test protocols be the part of this command?

Reply:
The command should accept the folder path against which to run, so we don;'t have to hardcode any paths!